### ISSUE TITLE
Support JsonPath wildcard "$.*.author" and recursion "$..author" in whenIgnoringPath

### ISSUE THREAD

**Issue by RobertDiebels** [lukas-krecan/JsonUnit#198]
**Read the documentation, maybe we already have the feature**

**Is your feature request related to a problem? Please describe.**
Limited implementation of JSONPath.

**Describe the solution you'd like**
The JSONPath matcher seems to be limited to the use of an any-element-in-array matcher (`[*].`). 
We'd like a any-key-in-object matcher (`$.*.author`) and a recursive matcher (`$..author`). This would allow us to ignore keys at our desired depths without having to specify the exact keys where a `author` key might be present.

**Describe alternatives you've considered**
Specifying the exact keys for each depth where a `author` key might be present.

**Additional context**
None.

**Comment by lukas-krecan** [comment]
Thanks, will look at it

**Comment by sugyan84** [comment]
Yeah, even I need all capabilities of JSONPath, as I have included this library into one of my project. And to generalize any JSON matching, I need the JSONPath things to be of full capability, if possible, or else atleast the path matching full capability.

**Comment by sugyan84** [comment]
I'm trying with this form of JsonPath. 
"$['proxy_gops_get_risk_evaluation_results_riskselfhelpserv']['output']"

But, when I specify a JsonPath for ignoring, I see it uses combined two forms of JsonPath, and hence it says, path is not found, and doesn't ignore it.

```
String jsonPath1 = "$['proxy_gops_get_risk_evaluation_results_riskselfhelpserv']['output']";
String ignoredPath = "$['proxy_gops_get_risk_evaluation_results_riskselfhelpserv']['output']['evaluations'][*]['decisions'][*]['decision_time']";
```

```
assertThatJson(inPath(jsonString, jsonPath1))
                    .when(Option.IGNORING_ARRAY_ORDER, TREATING_NULL_AS_ABSENT)
                    .whenIgnoringPaths(ignoredPath)
                    .isEqualTo(inPath(jsonString, jsonPath2));
```

**Comment by sugyan84** [comment]
![two-denotions](https://user-images.githubusercontent.com/33655437/66207621-d7d29900-e6d0-11e9-9fe0-409e167ddd3e.PNG)

**Comment by lukas-krecan** [comment]
Yes, the feature is not supported yet. You can use proxy_gops_get_risk_evaluation_results_riskselfhelpserv.output.evaluations[*].... syntax which should work

**Comment by lukas-krecan** [comment]
Released in 2.10.0 - supported only in AssertJ API

**Comment by RobertDiebels** [comment]
@lukas-krecan Great! Thanks a bunch!
